R/pivot_wider.R

```diff
@@ -96,9 +91,16 @@ pivot_wider..tidytable <- function(.df,
     glue_df <- distinct.(.df, !!!syms(names_from))
     values_from_reps <- nrow(glue_df)
     glue_df <- vec_rep(glue_df, length(values_from))
-    glue_df$.value <- vec_rep_each(values_from, values_from_reps)
+    glue_df <- mutate.(glue_df,
+                       .value = vec_rep_each(values_from, values_from_reps),
+                       .before = 1)
 
     glue_vars <- as.character(glue_data(glue_df, names_glue))
+    # mimic column names assigned by data.table::dcast()
+    if (length(values_from) <= 1) {
+      glue_df[[".value"]] <- NULL
+    }
+    names(glue_vars) <- do.call(paste, c(glue_df, list(sep = names_sep)))
```

Comment:
Can we use `exec()` here? `do.call()` with extra arguments passed in a `list()` is a bit hard to read IMO.

``` r
names(glue_vars) <- exec(paste, !!!glue_df, sep = names_sep)
```

Reply:
Agreed! I've tested it now, exec and unquoting add only negligible constant overhead of about 10-15µs.